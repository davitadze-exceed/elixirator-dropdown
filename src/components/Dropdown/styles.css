.dropdown-root {
	display: flex;
	flex-direction: column;
	width: 200px;
}

.dropdown-root input {
	width: 100%;
	box-sizing: border-box;
}

.dropdown-container {
	height: 200px;
	overflow: auto;
}

.dropdown-option {
	cursor: pointer;
}

.dropdown-option:hover {
	background-color: gray;
}
