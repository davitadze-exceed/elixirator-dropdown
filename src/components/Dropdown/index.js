import React, { useState } from 'react';
import './styles.css';


const Dropdown = ({ values }) => {
  const [dropdownValue, setDropdownValue] = useState('');
  const [searchValue, setSearchValue] = useState('');
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);

  const handleDropdownClick = () => {
    setIsDropdownOpen(!isDropdownOpen);
  };

  const setSelectedValue = (value) => {
    setDropdownValue(value);
    setSearchValue('');
    setIsDropdownOpen(false);
  };

  const filteredOptions = searchValue ? values.filter(value => value.includes(searchValue)) : values;

  return <div className="dropdown-root">
    <button onClick={handleDropdownClick}>{dropdownValue || 'Dropdown'}</button>
    {
      isDropdownOpen && <div>
        <input
          type="text"
          onChange={(e) => setSearchValue(e.target.value)}
          value={searchValue}
        />
        <div className="dropdown-container">
          {
            filteredOptions.map(value => <div className="dropdown-option" key={`dropdown-${value}`}
                                              onClick={() => setSelectedValue(value)}>{value}</div>)
          }
          {
            !filteredOptions.length && <div>No matched values</div>
          }
        </div>
      </div>
    }
  </div>
};

export default Dropdown;
