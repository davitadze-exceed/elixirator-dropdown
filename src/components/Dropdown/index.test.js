import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Dropdown from './index';

test('renders "dropdown"', () => {
  const { getByText } = render(<Dropdown />);
  const linkElement = getByText(/dropdown/i);
  expect(linkElement).toBeInTheDocument()
});

describe('Dropdown', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Dropdown values={[]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
