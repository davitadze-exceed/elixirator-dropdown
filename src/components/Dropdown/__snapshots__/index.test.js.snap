// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<div
  className="dropdown-root"
>
  <button
    onClick={[Function]}
  >
    Dropdown
  </button>
</div>
`;

exports[`Dropdown snapshot renders 1`] = `
<div
  className="dropdown-root"
>
  <button
    onClick={[Function]}
  >
    Dropdown
  </button>
</div>
`;
